### PR TITLE
Sk public to hex

### PR DESCRIPTION
**Pull request by SergeyPl4ik**

### CODE REVIEW

File: src/modules/dapps.js [hunk 1]
```diff
@@ -1524,9 +1531,10 @@ DApps.prototype.internal = {
 
             const hash = crypto.createHash('sha256').update(req.body.secret, 'utf8').digest();
             const keypair = library.ed.makeKeypair(hash);
+			const publicKey = keypair.publicKey.toString('hex');
```
Comment (alex-senyk) [inline src/modules/dapps.js]: Alignment (not critical)

File: src/modules/dapps.js [hunk 2]
```diff
@@ -1543,6 +1551,9 @@ DApps.prototype.internal = {
                         if (account.multisignatures.indexOf(keypair.publicKey.toString('hex')) < 0) {
                             return setImmediate(cb, 'Account does not belong to multisignature group');
                         }
+						if (account.multisignatures.indexOf(publicKey) < 0) {
```
Comment (alex-senyk) [inline src/modules/dapps.js]: Alignment

File: src/sql/delegates.js [hunk 3]
```diff
@@ -36,7 +36,7 @@ const DelegatesSql = {
             'FROM delegates d',
             'LEFT JOIN mem_accounts m ON d.username = m.username',
             'LEFT JOIN trs t ON d."transactionId" = t.id',
-            'LEFT JOIN (SELECT "dependentId", COUNT(1)::int AS voters_cnt from mem_accounts2delegates GROUP BY "dependentId") v ON v."dependentId" = ENCODE(m."publicKey", \'hex\')',
+            'LEFT JOIN (SELECT "dependentId", COUNT(1)::int AS voters_cnt from mem_accounts2delegates GROUP BY "dependentId") v ON v."dependentId" = m."publicKey"',
```
Comment (alex-senyk) [inline src/sql/delegates.js]: count(1)::int 
is redundant

File: src/sql/migrations/20190114181040_mem_accounts_publicKey.sql [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      b."numberOfTransactions" AS "b_numberOfTransactions",
+      b."totalAmount" AS "b_totalAmount",
+      b."totalFee" AS "b_totalFee",
+      b."reward" AS "b_reward",
+      b."payloadLength" AS "b_payloadLength",
+      m."username" AS "m_username",
+      ENCODE(b."payloadHash", 'hex') AS "b_payloadHash",
+      ENCODE(b."generatorPublicKey", 'hex') AS "b_generatorPublicKey",
+      ENCODE(b."blockSignature", 'hex') AS "b_blockSignature",
+      (SELECT MAX("height") + 1 FROM blocks) - b."height" AS "b_confirmations"
```
Comment (alex-senyk) [inline src/sql/migrations/20190114181040_mem_accounts_publicKey.sql]: This is possible loss of performance, using "with" query may help.

File: src/sql/migrations/20190114181040_mem_accounts_publicKey.sql [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+DROP VIEW IF EXISTS blocks_list;
+DROP INDEX "mem_accounts_get_delegates";
+
+ALTER TABLE "mem_accounts"
+ ALTER COLUMN "publicKey" TYPE VARCHAR(64) USING ENCODE("publicKey",'hex');
+
+CREATE INDEX IF NOT EXISTS "mem_accounts_get_delegates" ON "mem_accounts" ("vote" DESC, "publicKey" ASC) WHERE "isDelegate" = 1;
+
+CREATE VIEW blocks_list AS
```
Comment (Verbalist) [inline src/sql/migrations/20190114181040_mem_accounts_publicKey.sql]: remove

**Pull request merged**